src/OrbitBase/TaskGroupTest.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gtest/gtest.h>
+#include <stddef.h>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/TaskGroup.h"
+#include "OrbitBase/ThreadPool.h"
+#include "OrbitBase/UniqueResource.h"
+
+using orbit_base::TaskGroup;
+using orbit_base::ThreadPool;
+
+namespace {
+
+void ShutdownThreadPool(std::shared_ptr<ThreadPool> thread_pool) { thread_pool->ShutdownAndWait(); }
+
+[[nodiscard]] ThreadPool* GetTestThreadPool() {
+  constexpr size_t kNumThreads = 4;
+  constexpr absl::Duration kThreadTtl = absl::Milliseconds(5);
+  static orbit_base::unique_resource handle{
```

Comment:
Shouldn't that be a member variable of a test fixture and set the the test's start up.

But I also don't feel strong about that.